### PR TITLE
feat: implement MarkdownOutput component

### PR DESCRIPTION
**Pull request by wildergd**
fix: #1641 

## Changes proposed in this PR:
- add MarkdownOutput component 

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

**Comment by wildergd** [comment]
There are some commented lines ... this is still WIP....

File: src/components/MarkdownOutput/renderers/styled/index.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    background-color: ${props => props.palette.background.main};
+    border-spacing: 0;
+    width: 100%;
+    box-sizing: border-box;
+    border-radius: 10px;
+    border: 1px solid ${props => props.palette.border.divider};
+`;
+
+export const TableHead = attachThemeAttrs(styled.thead)`
+`;
```
Comment (yvmunayev) [inline src/components/MarkdownOutput/renderers/styled/index.js:88]: check this

File: src/components/MarkdownOutput/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export interface MarkdownOutputProps extends BaseProps {
+    id?: string;
+    renderers?: object;
+    linkTarget?: string;
+    transformLinkUri?: () => void;
+    transformImageUri?: () => void;
+    plugins?: object[];
+    parserOptions?: object;
+}
```
Comment (yvmunayev) [inline src/components/MarkdownOutput/index.d.ts:12]: add source prop

**Comment by yvmunayev** [comment]
Screen Shot 2020-07-01 at 4 00 55 AM
check this

**Comment by yvmunayev** [comment]
markdown
check this

**Comment by maxxgreene** [comment]
@wildergd check this code sandbox I think we can reduce considerably the complexity of the component 
https://codesandbox.io/s/stoic-leaf-sk4ks?file=/src/App.js

**Comment by wildergd** [comment]
I tried that... and yest is quite simple and that's true.. 2 lines of code.. the main problem i found  in `remark-react` is that `options.remarkReactComponents` is not quite well documented so it is not clear what props are passed to react components... 
about markdown abstract-syntax-tree ... there is a lot of documentation about it... you can easily parse it and map element attributes (alignment, heading levels, etc) to component props... renderers are based on component types instead of html tag...

**Comment by maxxgreene** [comment]
keep it simple with `remark-react`, check the `HTML` generated and provide a component for each element, by default `remark-react` will generate plain `HTML` element ... just write styled-components for each element that obey our theme implementation.

**Comment by wildergd** [comment]
Ok.. code refactored... more simple now.. but styling is a bit more complicated now... for example... <li>.... you cannot differentiate if is is a listItem in a simple list.. or an ordered one... or a task list... what if we want to hide the listitem marker on a task list?... the listitem is blind.... it doesn't know the list type... and css does not allow you to style a parent from a child... so you cannot use the checkbox and from there set style on it's parent listitem ot list...  ...

File: package.json [hunk 3]
```diff
@@ -36,7 +36,12 @@
         "clipboard-copy": "^2.0.0",
         "prop-types": "^15.6.2",
         "react-async-script-loader": "^0.3.0",
+        "remark": "^12.0.0",
+        "remark-parse": "^8.0.2",
+        "remark-react": "^7.0.1",
         "styled-components": "^4.3.2",
+        "unified": "^9.0.0",
+        "unist-util-visit": "^2.0.2",
```
Comment (reiniergs) [inline package.json:44]: remove `remark` and `unist-util-visit` since we don't need it anymore

File: src/components/MarkdownOutput/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { ReactNode } from 'react';
+import { BaseProps } from '../types';
+
+export interface MarkdownOutputProps extends BaseProps {
+    id?: string;
+    renderers?: object;
+    plugins?: object[];
+    parserOptions?: object;
+    source: string;
```
Comment (reiniergs) [inline src/components/MarkdownOutput/index.d.ts:9]: - source => value
- remove renderers , plugins, parserOptions(we wont allow that for now) 
- source is not required(by default empty string)

File: src/components/MarkdownOutput/index.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        id,
+        className,
+        style,
+        source: sourceInProps,
+        renderers: customRenderers,
+        parserOptions,
+        plugins,
+    } = props;
+
+    const source = sourceInProps || '';
```
Comment (reiniergs) [inline src/components/MarkdownOutput/index.js:17]: use defaultProps

File: src/components/MarkdownOutput/index.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    const result = useMarkdownToReact(source, parserOptions, plugins, {
+        ...defaultRenderers,
+        ...customRenderers,
+    });
```
Comment (reiniergs) [inline src/components/MarkdownOutput/index.js:21]: it will be just
```js
const result = useMarkdownToReact(value);
```

**Pull request merged**